server/service/src/number.rs

```diff
@@ -8,9 +11,74 @@ pub fn next_number(
     // Should be done in transaction
     let next_number = connection.transaction_sync(|connection_tx| {
         let repo = NumberRowRepository::new(&connection_tx);
-        let next_number = repo.get_next_number_for_type_and_store(r#type, store_id)?;
-
-        Ok(next_number.number)
+        let number = repo.find_one_by_type_and_store(r#type, store_id)?;
+
+        if number.is_some() {
+            let next_number = repo.get_next_number_for_type_and_store(r#type, store_id, None)?;
+            Ok(next_number.number)
```

Comment:
:nit early return

Reply:
Not entirely sure what the request is here, is it for `return Ok(next_number.number);`? I'll go for it with the other PR - it's consistent with the match statement which uses `return` syntax. FWIW it's returning evaluated value into another variable (that it shadows):

![image](https://user-images.githubusercontent.com/7684221/208318296-79d6a3c4-61c3-4938-aa2e-0ea39435bd37.png)

Reply:
I was meaning something like this: https://github.com/openmsupply/open-msupply/compare/%23811-Number-records-removed-from-sync-...811-refactor-example

Reply:
Oh heck yea that's way shorter! Thanks